discovery_engine/lib/src/domain/engine/mock_engine.dart

```diff
@@ -103,8 +103,8 @@ class MockEngine implements Engine {
 
   @override
   Future<void> setMarkets(
-    List<HistoricDocument> history,
-    List<FeedMarket> markets,
+    final List<HistoricDocument> history,
```

Comment:
We decided to not enable the linter to enforce final parameters so they should stay as they are.

Reply:
actually they are already `final` in the `Engine` (or more precisely in the `DiscoveryEngineFfi` implementation of the `Engine`) and this just aligns the `MockEngine`.

since we have a mixture of `final` and non-`final` arguments all over the code base, how should we proceed?

Reply:
I would say:

- not add `final` to new parameters
- when touching a function with final parameters, remove the final
- do not remove final on functions in files which else-wise would not have been touched

Reply:
We decided to not force it, not to not used it. If we never want final on parameters we should check if there is a linter for that.